src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractResolver.cs

```diff
@@ -8,7 +8,7 @@ namespace System.Runtime.Serialization
 {
     public abstract class DataContractResolver
     {
-        public abstract bool TryResolveType(Type type, Type declaredType, DataContractResolver knownTypeResolver, out XmlDictionaryString typeName, out XmlDictionaryString typeNamespace);
-        public abstract Type ResolveName(string typeName, string typeNamespace, Type declaredType, DataContractResolver knownTypeResolver);
+        public abstract bool TryResolveType(Type type, Type? declaredType, DataContractResolver knownTypeResolver, out XmlDictionaryString? typeName, out XmlDictionaryString? typeNamespace);
```

Comment:
`typeName` and `typeNamespace` looked like good candidates to have`NotNullWhen(true)` but `KnownTypeDataContractResolver ` doesn't honor that:
https://github.com/dotnet/runtime/blob/bcfd90de2a96f4fbe84448df5989b7fd8b258c8a/src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/KnownTypeDataContractResolver.cs#L31-L35

Reply:
Correct. A great reason why these nullable annotations add value - without them the signature looks like every other "TryGet" pattern where the `out` param is non-null when `true` is returned. But here it is possible that it resolved the type, but still set `typeName` and `typeNamespace` to null.